docs/labs/redos.html

```diff
@@ -0,0 +1,159 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta http-equiv="X-UA-Compatible" content="IE=edge">
+<meta name="viewport" content="width=device-width, initial-scale=1">
+<link rel="stylesheet" href="https://best.openssf.org/assets/css/style.css">
+<link rel="stylesheet" href="checker.css">
+<script src="js-yaml.min.js"></script>
+<script src="checker.js"></script>
+<link rel="license" href="https://creativecommons.org/licenses/by/4.0/">
+
+<!-- See create_labs.md for how to create your own lab! -->
+
+<!-- Sample expected answer -->
+<script id="expected0" type="plain/text">
+  query('id').isLength({max:50}).matches( /^[A-Z0-9]+$/ ),
+</script>
+
+<!-- Full pattern of correct answer -->
+<script id="correct0" type="plain/text">
+\s* query \( ('id'|"id"|`id`) \)
+\. isLength \( \{ max: 50 ,? \} \)
+\. matches \( \/\^
+\[A-Z0-9\]\+\$\/ \) , \s*
```

Comment:
The line break between the last 2 lines will be interpreted as "arbitrary whitespace allowed here", but whitespace is *meaningful* in a regex by default, so you don't want to do that. Expressing regular expressions, using regular expressions, is kind of a worst case due to the backslashes, thanks for working it through!

I think it'd be better to replace the last two lines with this one line:

~~~~
\. matches \( \/\^\[A-Z0-9\]\+\$\/ \) , \s*
~~~~